lib/cs_guide/postcode_lat_long.ex

```diff
@@ -0,0 +1,151 @@
+defmodule CsGuide.PostcodeLatLong do
+  alias NimbleCSV.RFC4180, as: CSV
+  alias CsGuide.Resources.Venue
+  import Ecto.Query
+  use GenServer
+
+  @zip_file_name "postcodes.zip"
+  @zip_file_name_charlist String.to_charlist(@zip_file_name)
+  @postcode_file_charlist 'ukpostcodes.csv'
+
+  def start_link do
+    GenServer.start_link(__MODULE__, %{}, name: PostcodeCache)
+  end
+
+  def init(initial_state) do
+    case Mix.env do
+      :test ->
+        IO.puts("Skipping postcode storage")
+        # This can be changed in the future. Just didn't want to have to clone
+        # csv file everytime that we run tests. File is currently being used
+        # to run priv/repo/add_lat_long_to_venue.exs. If we build in
+        # functionality in the future that checks to see if entered postcodes
+        # (venue creation for example) are correct, then we can check them
+        # against the cache. For testing at that point we can create a smaller
+        # csv file which we can use to test with.
+      _ ->
+        postcode_cache = :ets.new(:postcode_cache, [:set, :protected, :named_table])
```

Comment:
@RobStallion it would be _amaze_ if you could port this to a "how to use ETS" example in https://github.com/dwyl/learn-elixir

Reply:
@nelsonic On it 😄